src/Features/Core/Portable/StackTraceExplorer/StackTraceExtensions.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Text;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.Common;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.StackFrame;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.StackTraceExplorer
+{
+    using StackFrameToken = EmbeddedSyntaxToken<StackFrameKind>;
+    using StackFrameTrivia = EmbeddedSyntaxTrivia<StackFrameKind>;
+
+    internal static class StackTraceExtensions
+    {
+        public static string CreateString(this StackFrameNode node, StackFrameTree tree)
+            => node.CreateString(tree, skipTrivia: false);
+
+        public static string CreateString(this StackFrameNode node, StackFrameTree tree, bool skipTrivia)
+        {
+            var span = skipTrivia
+                ? node.GetSpanWithoutTrivia(skipLeadingTrivia: true, skipTrailingTrivia: true)
```

Comment:
this doesn't really work.  imagine something as simple as ` X . Y `.  This will produce the span that contains `X . Y`, but the itnernal spaces will remain.  What would work would be to just recurse down the node and just add the token contents in order to a string builder.

Reply:
note: this woudl be fine to added to EmbeddedSyntaxNode/Token/etc.

Reply:
Do you mean to override `ToString` on `Embedded*`? Or should they be called `CreateString` to align with `VirtualChars`?